smart-contracts/wasm-chain-integration/src/utils.rs

```diff
@@ -153,31 +183,53 @@ impl machine::Host<ArtifactNamedImport> for TestHost {
                 std::str::from_utf8(&memory[filename_start..filename_start + filename_length])?
                     .to_owned();
             bail!(ReportError::Reported {
+                quickcheck: self.rng_used,
```

Comment:
Why do we copy this value out here?

Can we not just read  it from the testhost? What advantages does this have?

One issue I have with it is that the get_random may have been used before. It does not mean that this specific report_error has anything to do with quickcheck.

Reply:
Fixed, use `rng_used` of `TestHost` directly in `cargo-concordium`.